test/common/stats/stats_impl_test.cc

```diff
@@ -170,7 +158,32 @@ class DefaultTagRegexTester {
                        stat_name);
   }
 
-  std::vector<TagExtractorPtr> tag_extractors_;
+  /**
+   * Reimplements TagProducerImpl::produceTags, but extracts the tags in reverse order.
+   * His helps demonstrate that the order extractors does not matter to the end result,
+   * assuming we don't care about tag-order. This is in large part correct by design because
+   * stat_name is not mutated until all the extraction is done.
+   * @param metric_name std::string a name of Stats::Metric (Counter, Gauge, Histogram).
+   * @param tags std::vector a set of Stats::Tag.
+   */
```

Comment:
Nit: Add a `@return` doc string. Not normally required for tests, but since the input params are well documented might as well do it for return.

Reply:
done

Reply:
done